src/Styleguide/Components/Lightbox/Lightbox.tsx

```diff
@@ -0,0 +1,286 @@
+  }
+
+  renderLightbox() {
+    const { slider } = this.state
+    return (
+      <FadeTransition
+        in={this.state.shown}
+        mountOnEnter
+        unmountOnExit
+        timeout={{ enter: 250, exit: 300 }}
+      >
+        <DeepZoomContainer
+          onMouseMove={this.detectActivity}
+          onWheel={this.detectActivity}
+          onTouchStart={this.detectActivity}
+          onTouchMove={this.detectActivity}
+          innerRef={this.state.deepZoomRef as any}
+        >
+          <Box
+            position="absolute"
+            top={space(3) / 2}
+            right={space(3) / 2}
+            zIndex={1001}
+          >
+            <CloseButton onClick={() => this.hide()} />
+          </Box>
+          <Flex
+            position="absolute"
+            width="100%"
+            justifyContent="center"
+            zIndex={1001}
+            bottom={space(2)}
+          >
+            <FadeTransition
+              in={this.state.showZoomSlider}
+              timeout={{ enter: 50, exit: 150 }}
+            >
+              <Slider
+                min={slider.min}
+                max={slider.max}
+                step={slider.step}
+                value={slider.value}
+                onChange={this.onSliderChanged}
+                onZoomInClicked={() => this.zoomIn()}
+                onZoomOutClicked={() => this.zoomOut()}
+              />
+            </FadeTransition>
+          </Flex>
+        </DeepZoomContainer>
+      </FadeTransition>
+    )
+  }
+
+  renderPortal = () => {
+    return this.state.element
+      ? ReactDOM.createPortal(this.renderLightbox(), this.state.element)
+      : null
+  }
+
+  render() {
+    const { children } = this.props
+    const modifiedChildren = React.Children.map(children, child =>
+      React.cloneElement(child as any, {
+        style: { cursor: "zoom-in" },
+        onClick: this.show,
+      })
+    )
+    return (
+      <React.Fragment>
+        {this.renderPortal()}
+        {modifiedChildren}
+      </React.Fragment>
+    )
+  }
+
+  postRender = () => {
+    this.state.viewer.addHandler(
+      "zoom",
+      bind(throttle(this.onZoomChanged, 50), this)
+    )
+    this.state.viewer.addHandler(
+      "tile-drawn",
+      once(() => {
+        this.setState({
+          slider: {
+            ...this.state.slider,
+            min: this.state.viewer.viewport.getMinZoom(),
+            max: this.state.viewer.viewport.getMaxZoom(),
+            value: this.state.viewer.viewport.getHomeZoom(),
+          },
+        })
+      })
+    )
+    this.detectActivity()
+  }
+
+  show = event => {
+    this.setState({ shown: true, showZoomSlider: true })
+  }
+
+  hide = () => {
+    this.setState({ shown: false })
+    if (this.state.viewer) {
+      this.state.viewer.destroy()
+      this.state.viewer = null
+    }
+    document.removeEventListener(KEYBOARD_EVENT, this.handleKeyPress)
+    clearTimeout(this.state.activityTimer)
```

Comment:
Ah yep, that answers my earlier question, you mean a ‘timer ID’. You only get a reference to the timers.

Reply:
Ah, gotcha. What's the type of that?